drivers/adc/ad4858/ad4858.c

```diff
@@ -571,8 +571,14 @@ int ad4858_set_chn_or_limit(struct ad4858_dev *dev, uint8_t chn,
 	if (!dev || (chn >= AD4858_NUM_CHANNELS))
 		return -EINVAL;
 
-	/* OR limit value is spanned over bits [24:4] */
-	ret = ad4858_reg_write(dev, AD4858_REG_CH_OR(chn), (or_limit << 4));
+	if (dev->prod_id == AD4855_PROD_ID_L) {
```

Comment:
the ad4858 driver supports a a lot of parts. I'd go for `switch..case` statements throughout this patch. It's easier to extend in the future.